homeassistant/components/snapcast/__init__.py

```diff
@@ -1 +1,86 @@
 """The snapcast component."""
+import logging
+import socket
+
+
+import voluptuous as vol
+
+import snapcast.control
+from snapcast.control.server import CONTROL_PORT
+
+from homeassistant.helpers.discovery import async_load_platform
+
+from homeassistant.helpers import config_validation as cv, entity_platform
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PORT,
+)
+
+from .const import (
+    ATTR_LATENCY,
+    ATTR_MASTER,
+    CLIENT_PREFIX,
+    CLIENT_SUFFIX,
+    DATA_KEY,
+    GROUP_PREFIX,
+    GROUP_SUFFIX,
+    SERVICE_JOIN,
+    SERVICE_RESTORE,
+    SERVICE_SET_LATENCY,
+    SERVICE_SNAPSHOT,
+    SERVICE_UNJOIN,
+    GROUP_DISABLE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+#   {vol.Required(CONF_HOST): cv.string, vol.Optional(CONF_PORT): cv.port}
+# )
+
+CONFIG_SCHEMA = vol.Schema(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

We can import the existing platform config yaml in the media_player platform to a config entry via the config flow. See eg the nut sensor platform:
https://github.com/home-assistant/core/blob/99318b7b11080f3e4cc9644b660d935d52cec7c0/homeassistant/components/nut/sensor.py#L60-L67

Reply:
I suggest we do this refactor first in a separate PR before adding any new features.

Reply:
I'm  a bit confused. The configuration should not be in the media_player or in the sensor. 
I want the host be defined like this:

```
snapcast:
    host: 192.168.1.10
```
because the linking to the snapcast host is now in the common part for media_player or sensor...

Reply:
Yes, but we don't allow changes to config yaml anymore. The configuration will be done via a config flow which creates a config entry.

To not break existing users we can do the import of the old media_player platform config yaml to the config flow and config entry. New users will set up the integration via the integrations menu in the GUI.